lib/ood_core/job/adapters/slurm.rb

```diff
@@ -147,6 +197,62 @@ def submit_string(str, args: [], env: {})
             call("sbatch", *args, env: env, stdin: str.to_s).strip.split(";").first
           end
 
+          # Fields requested from a formatted `squeue` call
+          # Note that the order of these fields is important
+          def all_squeue_fields
+            {
+              account: "%a",
+              job_id: "%A",
+              exec_host: "%B",
+              min_cpus: "%c",
+              cpus: "%C",
+              min_tmp_disk: "%d",
+              nodes: "%D",
+              end_time: "%e",
+              dependency: "%E",
+              features: "%f",
+              array_job_id: "%F",
+              group_name: "%g",
+              group_id: "%G",
+              over_subscribe: "%h",
+              sockets_per_node: "%H",
+              array_job_task_id: "%i",
+              cores_per_socket: "%I",
+              job_name: "%j",
+              threads_per_core: "%J",
+              comment: "%k",
+              array_task_id: "%K",
+              time_limit: "%l",
+              time_left: "%L",
+              min_memory: "%m",
+              time_used: "%M",
+              req_node: "%n",
+              node_list: "%N",
+              command: "%o",
+              contiguous: "%O",
+              qos: "%q",
+              partition: "%P",
+              priority: "%Q",
+              reason: "%r",
+              start_time: "%S",
+              state_compact: "%t",
+              state: "%T",
+              user: "%u",
+              user_id: "%U",
+              reservation: "%v",
+              submit_time: "%V",
+              wckey: "%w",
+              licenses: "%W",
+              excluded_nodes: "%x",
+              core_specialization: "%X",
+              nice: "%y",
+              scheduled_nodes: "%Y",
+              sockets_cores_threads: "%z",
+              work_dir: "%Z",
+              gres: "%b",  # must come at the end to fix a bug with Slurm 18
```

Comment:
Tangential to this review: an outcome of the last ops meeting was the revelation that `gres`/`tres` is how Slurm allows GPUs to be a Trackable Resource. We may end up caring more about this field in the future.

Reply:
Oh boy!